src/uct/rocm/base/rocm_base.c

```diff
@@ -183,10 +183,18 @@ ucs_status_t uct_rocm_base_detect_memory_type(uct_md_h md, const void *addr,
     }
 
     info.size = sizeof(hsa_amd_pointer_info_t);
-    status = hsa_amd_pointer_info((void*)addr, &info, NULL, NULL, NULL);
+    status    = hsa_amd_pointer_info((void*)addr, &info, NULL, NULL, NULL);
     if ((status == HSA_STATUS_SUCCESS) &&
-        (info.type != HSA_EXT_POINTER_TYPE_UNKNOWN)) {
-        *mem_type_p = UCS_MEMORY_TYPE_ROCM;
+        (info.type == HSA_EXT_POINTER_TYPE_HSA)) {
+        hsa_device_type_t dev_type;
```

Comment:
better move this to the beginning of the function

Reply:
done